v2/yarpcthrift/register.go

```diff
@@ -71,21 +65,20 @@ func BuildProcedures(s Service, opts ...RegisterOption) []yarpc.TransportProcedu
 		proto = rc.Protocol
 	}
 
-	rs := make([]yarpc.TransportProcedure, 0, len(s.Methods))
+	rs := make([]yarpc.EncodingProcedure, 0, len(s.Methods))
 
 	for _, method := range s.Methods {
-		var spec yarpc.TransportHandlerSpec
-		spec = yarpc.NewUnaryTransportHandlerSpec(unaryTransportHandler{
-			ThriftHandler: method.Handler,
-			Protocol:      proto,
-			Enveloping:    rc.Enveloping,
-		})
-
-		rs = append(rs, yarpc.TransportProcedure{
+		var spec yarpc.EncodingHandlerSpec
+		spec = yarpc.NewUnaryEncodingHandlerSpec(method.Handler)
+		codec := func() yarpc.InboundCodec {
+			return newCodec(proto, rc.Enveloping)
+		}
+		rs = append(rs, yarpc.EncodingProcedure{
 			Name:        yarpcprocedure.ToName(s.Name, method.Name),
 			HandlerSpec: spec,
 			Encoding:    Encoding,
 			Signature:   method.Signature,
+			Codec:       codec,
```

Comment:
Great to see that the separation of the `Codec` and the `Handler` actually ended up being clean / not invasive!